File: 2412-minimum-money-required-before-transactions/2412-minimum-money-required-before-transactions.java
class Solution {
    public long minimumMoney(int[][] transactions) {
        long minMoney = 0; int minTransTillNow = 0;
        for(int[] transaction : transactions) {
            minTransTillNow = Math.max(minTransTillNow, Math.min(transaction[0], transaction[1]));
            minMoney += Math.max(transaction[0] - transaction[1], 0);
        }
        return minTransTillNow + minMoney;
    }
}

File: 2412-minimum-money-required-before-transactions/2412-minimum-money-required-before-transactions.py
class Solution:
    def minimumMoney(self, transactions: List[List[int]]) -> int:
        minTransTillNow, minMoney = 0, 0
        for transaction in transactions:
            minTransTillNow = max(minTransTillNow, min(transaction[0], transaction[1]))
            minMoney += max(transaction[0] - transaction[1], 0)
        return minTransTillNow + minMoney